src/x509.rs

```diff
@@ -48,7 +48,7 @@ impl<'a> X509Extension<'a> {
     }
 
     /// Return the extension type or `None` if the extension is not implemented.
-    pub fn parsed_extension(&self) -> &ParsedExtension {
+    pub fn parsed_extension(&self) -> &ParsedExtension<'a> {
```

Comment:
I'd like to understand - the compiler was inferring a wrong lifetime without the explicit one?

Reply:
Indeed! Lifetime elision was picking up the lifetime on the &self reference, and not the 'a lifetime on the instance.

My usage goes like this:
I have a long-lived ('a) object that carries the entire DER of the certificate.
A function parses it and returns &str data that is sliced from the same DER.
The &str should be able to live as long as the parent DER, but the X509Extension and ParsedExtension structs are local to the function; having the reference lifetime (function local) distinct from the inner 'a lifetime allows me to return the right &'a str.

Reply:
Ah, I see now, thank you for the details.
I also encountered this kind of limitations within DER, where the compiler did not apply "transitive" lifetimes to inner objects when returned, but agrees when I give it explicitly - maybe I will be able to fix this now!